Documentation/ApiOverview/ContentSecurityPolicy/Index.rst

```diff
@@ -46,7 +59,184 @@ need to be enabled.
+
+
+..  _content-security-policy-terminology:
+
+Terminology
+===========
+
+This document will use very specific wording that is part of the CSP W3C specification,
+these terms are not "invented" by TYPO3. Since reading the W3C RFC can be very
+intimidating, here are a few key concepts.
+
+..  note::
+    Skip to the section :ref:`content-security-policy-example` to see a "real-life" usage
+    scenario, if you can better understand from actual code examples.
+
+..  _content-security-policy-terminology-directives:
+
+Directives
+----------
+
+*   CSP consists of multiple rules (or "directives"), that are part of a "policy". This
+    policy says, what functionality the site's output is allowed to use.
+
+*   With that, several HTML tags can be controlled, like from which URLs images can be
+    requested from, if and from where iframes are allowed, if and from where JavaScripts
+    are allowed and so on. There is a long list of applicable directives, see
+    `https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy#directives`__
+    with specific identifiers like `default-src`, `img-src` and so on.
+
+*   Each directive may have several "attributes" (like the allowed URLs).
+
+*   Directives may build upon each other, a bit like CSS definitions (`Cascading Style Sheet`)
+    do. However, these are more meant to modify a basic rule on an earlier level,
+    and called "mutations". The relation of a "child rule" to it's "parent" is
+    also called "ancestor chain".
+
+..  _content-security-policy-terminology-policy:
+
+Applying the policy
+-------------------
+
+*   A final policy is compiled of all these directives, and then sent as a HTTP
+    response header `Content-Security-Policy: ...`.
+
+*   In TYPO3, directives can be specified via PHP syntax (within Extensions) and
+    YAML syntax (within site configuration).
+
+..  _content-security-policy-terminology-mutations:
+
+Mutations
+---------
+
+*   These rules can influence each other, this is where the concept of "mutations" come in.
+    The "policy builder" of TYPO3 applies each configured mutation, no matter where it was
+    defined.
+
+*   Because of this, each mutation (directive definition) needs a specific "mode" that can
+    instruct, how this mutation is applied: Should an existing directive be
+    `appended`, `removed` or `added` to the final policy.
+
+*   Each directive is then applied in regard to its defined mode and can list one or more
+    "sources" with the values of additional parameters of a directive. Sources are
+    web site addresses / URLs.
+
+..  _content-security-policy-terminology-nonces:
+
+Nonces
+------
+
+*   There are possible exemptions to directives for specific content created on specific
+    pages created by TYPO3 in your frontend (or backend modules). To verify, that these
+    exemptions are valid in a policy, a so-called "Nonce" (unique hash, a "**n**umber used **once**")
```

Comment:
A nonce is **not** a hash, , please avoid this term here, as you can also (by now not in TYPO3) use hashes (sha-256, sha-384, sha-512). It is generated from a cryptographically secure random number generator (like random_bytes())

Reply:
Very much so, thanks for pointing this unspecificity out.